### PR TITLE
Add deleteObjectsOnCompletion to spec

### PR DESCRIPTION
**Pull request by rcarrillocruz**
This will delete ACM objects on upgrade completion if set

### CODE REVIEW

File: controllers/clustergroupupgrade_controller.go [hunk 1]
```diff
@@ -224,20 +224,24 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
...
-		}
-		r.deletePolicies(ctx, clusterGroupUpgrade)
-		if err != nil {
-			return ctrl.Result{}, err
+		if clusterGroupUpgrade.Spec.DeleteObjectsOnCompletion {
+			err := r.deletePlacementRules(ctx, clusterGroupUpgrade)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			r.deletePlacementBindings(ctx, clusterGroupUpgrade)
```
Comment (irinamihai) [inline controllers/clustergroupupgrade_controller.go:237]: err :=
?

File: controllers/clustergroupupgrade_controller.go [hunk 2]
```diff
@@ -224,20 +224,24 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 					Message: "The ClusterGroupUpgrade CR has all upgrade policies compliant",
 				})
 			}
+		} else if readyCondition.Reason == "UpgradeTimedOut" {
+
```
Comment (irinamihai) [inline controllers/clustergroupupgrade_controller.go:228]: Was wondering if we could add a Log here, just to get an idea of what's happening. Your choice if it makes sense.

File: controllers/clustergroupupgrade_controller.go (hunk 1 above)
Comment (rcarrillocruz) [reply controllers/clustergroupupgrade_controller.go:237]: yep, you are right, thx!

**Review by yrobla — APPROVED** [review]

**Pull request merged**